mlflow/store/artifact/s3_artifact_repo.py

```diff
@@ -80,7 +81,7 @@ def _cached_get_s3_client(
 
 
 def _get_s3_client(
-    addressing_style="path",
+    addressing_style=MLFLOW_BOTO_CLIENT_ADDRESSING_STYLE.get(),
```

Comment:
We should have a default of None here since I believe this will only be evaluated once at import time, and look up the value from the env variable within the method

Reply:
Fixed.